### PR TITLE
Adding Shasta

### PR DESCRIPTION
**Pull request by NatPRoach**
FOR CONTRIBUTOR:
* [X] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [X] - License permits unrestricted use (educational + commercial)
* [X] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

**Comment by NatPRoach** [comment]
Realizing just now that there are still a few parameters in this that I forgot to add complete label / help text for. Will modify those and make a new commit shortly.

**Review by bgruening — COMMENTED** [review]
@NatPRoach please add also  .shed.yml file to trigger the tests.

**Comment by NatPRoach** [comment]
Will do, do you think this belongs in the Nanopore section or the Assembly section of the toolshed?

**Comment by bgruening** [comment]
Do both, you can list many categories if you want.

**Comment by NatPRoach** [comment]
Note: I added a few lower bounds for a few parameters based on reading through the arg parsing code of shasta, but figuring out the bounds for all of the parameters may be difficult. There doesn't seem to be any robust checking system in their arg parsing block to ensure the values fall within acceptable ranges for downstream analysis. This means values we allow users to provide may break the program, but figuring out which values will break things may take some time and testing.

File: tools/shasta/macros.xml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        </citations>
+    </xml>
+    <!--
+        input
+    -->
+    <token name="@INIT_INPUT_READS@"><![CDATA[
+#for $counter, $input in enumerate($in_data)
+    #if $input.is_of_type("fasta.gz")
+        #set $ext = "fasta"
+        gzip -dcf $input > ./input_${counter}.${ext} &&
```
Comment (bgruening) [inline tools/shasta/macros.xml:32]: input should be quoted

File: tools/shasta/macros.xml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        input
+    -->
+    <token name="@INIT_INPUT_READS@"><![CDATA[
+#for $counter, $input in enumerate($in_data)
+    #if $input.is_of_type("fasta.gz")
+        #set $ext = "fasta"
+        gzip -dcf $input > ./input_${counter}.${ext} && 
+    #elif $input.is_of_type("fastq.gz","fastqsanger.gz")
+        #set $ext = "fastq"
+        gzip -dcf $input > ./input_${counter}.${ext} &&
```
Comment (bgruening) [inline tools/shasta/macros.xml:35]: quotes

File: tools/shasta/shasta.xml [hunk 3]
```diff
@@ -0,0 +1,832 @@
...
+                <option value="0">Random</option>
+                <option value="1">Random, excluding globally overenriched</option>
+                <option value="2">Random, excluding overenriched even in a single read</option>
+                <option value="3">Read from file</option>
+            </param>
+            <param argument="--Kmers.k" name="k" optional="true" type="integer" label="Length of marker k-mers (In run length space)"/>
+            <param argument="--Kmers.probability" name="probability" optional="true" type="float" label="Fraction k-mers used as a marker"/>
+            <param argument="--Kmers.enrichmentThreshold" name="enrichment_threshold" optional="true" type="float" label="Enrichment threshold for defining cutoff in Kmers.generationMethod that select based on enrichment"/>
+            <param argument="--Kmers.file" name="file" optional="true" type="data" format="txt,tsv" label="The file containing the k-mers to be used as markers, one kmer per line. Only used if the Kmers.generationMethod is 'Read from file'"/>
```
Comment (bgruening) [inline tools/shasta/shasta.xml:401]: tabular here as well?

**Review by bgruening — COMMENTED** [review]
This is great work! Thanks @NatPRoach just 2 tiny comments.

**Review by bgruening — APPROVED** [review]

**Comment by bgruening** [comment]
Merci @NatPRoach!

**Pull request merged**